graylog2-server/src/main/java/org/graylog/plugins/map/geoip/MaxMindIpLocationResolver.java

```diff
@@ -46,28 +46,23 @@ public MaxMindIpLocationResolver(@Assisted Timer resolveTime,
     @Override
     public Optional<GeoLocationInformation> doGetGeoIpData(InetAddress address) {
 
-        GeoLocationInformation info;
+        Optional<GeoLocationInformation> info;
         try (Timer.Context ignored = resolveTime.time()) {
             final CityResponse response = databaseReader.city(address);
             final Location location = response.getLocation();
             final Country country = response.getCountry();
             final City city = response.getCity();
 
-            info = GeoLocationInformation.create(
-                    location.getLatitude(), location.getLongitude(),
-                    country.getGeoNameId() == null ? "N/A" : country.getIsoCode(),
-                    country.getGeoNameId() == null ? "N/A" : country.getName(),
-                    city.getGeoNameId() == null ? "N/A" : city.getName(),// calling to .getName() may throw a NPE
-                    "N/A",
-                    "N/A");
-        } catch (IOException | GeoIp2Exception | UnsupportedOperationException e) {
-            info = null;
+            GeoLocationInformation gli = GeoLocationInformation.create(location, country, city, "N/A");
+            info = Optional.of(gli);
+        } catch (IOException | GeoIp2Exception | NullPointerException | UnsupportedOperationException e) {
+            info = Optional.empty();
             if (!(e instanceof AddressNotFoundException)) {
                 LOG.debug("Could not get location from IP {}", address.getHostAddress(), e);
                 lastError = e.getMessage();
             }
         }
```

Comment:
Just one small suggestion: I think the `Optional<GeoLocationInformation> info` variable declaration on line 49 can be avoided by returning the new optionals in-line:
```
        try (Timer.Context ignored = resolveTime.time()) {
            final CityResponse response = databaseReader.city(address);
            final Location location = response.getLocation();
            final Country country = response.getCountry();
            final City city = response.getCity();

            return Optional.of(GeoLocationInformation.create(location, country, city, "N/A"));
        } catch (IOException | GeoIp2Exception | NullPointerException | UnsupportedOperationException e) {
            if (!(e instanceof AddressNotFoundException)) {
                LOG.debug("Could not get location from IP {}", address.getHostAddress(), e);
                lastError = e.getMessage();
            }
            return Optional.empty();
        }
```

Reply:
Updated in the latest push.  I should note that having a single exit point from a function/method is the preferred approach/best practice.  

Personally I find method calls as method arguments such as `Optional.of(GeoLocationInformation.create(location, country, city, "N/A"))` difficult to read, in particular when the second (or any number that follows) has more than one argument.